app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,110 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
+  before_action :load_and_authorize_posts, except: [:create, :reply]
```

Comment:
The post cannot be authorised in isolation without any contextual information on the topic. This would mean that all anyone can specify in the ability is to either update or delete a post, and this applies to every user. So any user can delete or update all posts.